app/services/katello/pulp3/repository/generic.rb

```diff
@@ -0,0 +1,32 @@
+module Katello
+  module Pulp3
+    class Repository
+      class Generic < ::Katello::Pulp3::Repository
+        def copy_content_for_source(source_repository, _options = {})
+          copy_units_by_href(source_repository.files.pluck(:pulp_id))
+        end
+
+        def distribution_options(path)
+          {
+            base_path: path,
+            publication: repo.publication_href,
+            name: "#{generate_backend_object_name}"
+          }
+        end
+
+        def remote_options
```

Comment:
`remote_options` and `partial_repo_path` will need to be made generic as well. Right now it's specific to the File repo type. I'd say for this PR, `remote_options` can just return `common_remote_options` and `partial_repo_path` could be defined in `lib/katello/repository_types/python.rb`.  You could probably just leave it as an empty string until you do the syncing stuff.